### PR TITLE
added nxp OTFAD driver

### PR DESCRIPTION
**Pull request by mathias-landolt-loepfe**
Added OTFAD

why i had to add it here [soc/arm/nxp_imx/rt/Kconfig.defconfig.mimxrt1176_cm7](https://github.com/mathias-landolt-loepfe/zephyr/compare/main...mathias-landolt-loepfe:zephyr:add_otfad_driver?expand=1#diff-9417a09e80b09fedd6ea9ea8151b9f34a5a4837129c2450a3494d155276c13fa) instead of just here [soc/arm/nxp_imx/rt/Kconfig.soc](https://github.com/mathias-landolt-loepfe/zephyr/compare/main...mathias-landolt-loepfe:zephyr:add_otfad_driver?expand=1#diff-4e3a9ad8f603a3cc4fb0ef3384b7d5ac697e313dbab920588cb49be2d201bac8) doesnt make sense to me...

### CODE REVIEW

File: soc/arm/nxp_imx/rt/Kconfig.soc [hunk 1]
```diff
@@ -334,6 +334,8 @@ config SOC_MIMXRT1176_CM7
 	select HAS_MCUX_SRC_V2
 	select HAS_MCUX_IOMUXC
 	select HAS_SWO
+	select HAS_MCUX_OTFAD
+	select HAS_MCUX_XBAR
```
Comment (DerekSnell) [inline soc/arm/nxp_imx/rt/Kconfig.soc:338]: should be HAS_MCUX_XBARA

Comment (mathias-landolt-loepfe) [reply soc/arm/nxp_imx/rt/Kconfig.soc:338]: Thanks. Sorry for this stupid mistake.

**Comment by mathias-landolt-loepfe** [comment]
Now it works as it should.

I realized that someone added the HAS_ADC_ETC to just one of the 1064. 
What would you say? Is it useful to have this common feature of the series to the Kconfig.series instead adding it to every soc?

Similar with the OTFAD feature. This is afaik present in all RT11xx...

Is this kind of contribution worth giving upstream?

**Comment by DerekSnell** [comment]
Hi @mathias-landolt-loepfe ,

> I realized that someone added the HAS_ADC_ETC to just one of the 1064. 

I think what happened here is a community member added the ADC_ETC driver, and probably only added to the device they are using.  But yes, we can add to all the devices that have the ADC_ETC feature.

> What would you say? Is it useful to have this common feature of the series to the Kconfig.series instead adding it to every soc?
> 
> Similar with the OTFAD feature. This is afaik present in all RT11xx...

As you have seen, Kconfig.series is very brief, and does not select any peripheral features.  Kconfig.soc has all the features defined for each SoC.  That method also considers future devices in this family, that may not have all the same features.  For example, the RT10xx devices also sharing these files do not have OTFAD (they do have a similar feature, but different peripheral).  And while I expect every RT MCU will have an ADC, it is possible future devices may not use the same ADC_ETC peripheral.  So this is why these features are not selected in a common series file.

> Is this kind of contribution worth giving upstream?

Absolutely, we all value your contributions like this, and helping to improve Zephyr.  One of the best features of Zephyr Project, is the great contributions from the whole community, and from you too :)

Best regards

**Pull request merged**